File: 02_common_uses/03_user_types.cpp
#include <iostream>
class MyClass{
    public:
    void SetMyInt(int new_value){

        my_int_ = new_value;
    }

    void Bar(){

    }
    int public_int_;
    MyClass(int my_int, int public_int): my_int_ {my_int}, public_int_ {} {}

    friend std::ostream& operator<<(std::ostream& os, const MyClass& my_class){
        os << "MyClass(my_int_: " << my_class.my_int_ << ", public_int_: " << my_class.public_int_ << ")";
    return os;
    }

    private:
    int my_int_;
};

int main(){
    MyClass my_class {3, 2};
    my_class.public_int_ = 22;
    std::cout<<my_class<<std::endl;
    return 0;
}

File: 02_common_uses/03_user_types.rs
#[derive(Debug)]
struct MyStruct{
    pub public_int: i32,
    my_int: i32
}


impl MyStruct{
    pub fn new(public_int: i32, my_int_new: i32) ->Self{

        return Self {
            public_int, //shorthand notation 
            my_int: my_int_new
        }
    }
    fn bar(&self) {

    }
    pub fn set_my_int(&self, new_value: i32){
        self.my_int = new_value;

    }
}


fn main(){
    let my_struct = MyStruct::new(3,2);
    my_struct.bar();
    my_struct.public_int = 22;
    my_struct.set_my_int(new_value);
    println!("{}", my_struct);

}